docs/src/content/docs/basic-concepts.mdx

```diff
@@ -49,9 +50,28 @@ const AIGreeting = gsx.Component<GreetingProps, string>(async ({ name }) => {
 });
 ```
 
+Components can return structured objects containing other components. The pattern also works with array operations like `map` to process items in parallel.
+
+```tsx
+const AnalyzePosts = gsx.Component<AnalyzePostsProps, AnalyzePostsOutput>(
+  ({ posts }) => ({
+    analyses: posts.map((post) => ({
+      summary: <PostSummarizer post={post} />,
+      commentAnalysis: (
+        <CommentsAnalyzer postId={post.id} comments={post.comments} />
+      ),
+    })),
+  }),
+);
+```
+
+In this example, `AnalyzePosts` will produce an array of objects that contains the `summaries` and `commentAnalysis` for each post.
```

Comment:
Mention that the framework handles resolving all of the nested components and collecting them into a final value. No extra work required.

Reply:
added